ci/loadvet/main.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"go/ast"
+	"go/token"
+	"io"
+	"os"
+	"strings"
+	"unicode"
+
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/singlechecker"
+)
+
+func main() {
+	singlechecker.Main(loadAnalyzer)
+}
+
+var loadAnalyzer = &analysis.Analyzer{
+	Name: "loadvet",
+	Doc:  "reports ill-formatted prometheus load directives",
+	Run:  run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	for _, file := range pass.Files {
+		if !strings.HasSuffix(file.Name.String(), "_test") {
+			continue
+		}
+		var stack []ast.Node
+		ast.Inspect(file, func(n ast.Node) bool {
+			defer func() {
+				if n == nil {
+					stack = stack[:len(stack)-1]
+				} else {
+					stack = append(stack, n)
+				}
+			}()
+			if n == nil || len(stack) == 0 {
+				return true
+			}
+			parent := stack[len(stack)-1]
+			if _, ok := parent.(*ast.KeyValueExpr); !ok {
+				return true
+			}
+
+			s, ok := n.(*ast.BasicLit)
+			if !ok {
+				return true
+			}
+			if s.Kind != token.STRING {
+				return true
+			}
+			quote := s.Value[0]
+			if quote != '`' {
+				return true
+			}
+			cont := s.Value[1 : len(s.Value)-1]
+			if !strings.HasPrefix(cont, "load") {
```

Comment:
I think there are other directives as well like `clear` and `eval`: https://github.com/prometheus/prometheus/blob/main/promql/test.go#L252

Reply:
i was intending it only to format our tests for now; to do it properly we would need to be able to parse those load directives.